include/seastar/core/reactor.hh

```diff
@@ -371,20 +381,15 @@ private:
         void handle_signal_once(int signo, noncopyable_function<void ()>&& handler);
         static void action(int signo, siginfo_t* siginfo, void* ignore);
         static void failed_to_handle(int signo);
-    private:
-        struct signal_handler {
-            signal_handler(int signo, noncopyable_function<void ()>&& handler);
-            noncopyable_function<void ()> _handler;
-        };
-        std::atomic<uint64_t> _pending_signals;
-        std::unordered_map<int, signal_handler> _signal_handlers;
     };
-
+    
     signals _signals;
     std::unique_ptr<thread_pool> _thread_pool;
+    
     friend class thread_pool;
     friend class thread_context;
     friend class internal::cpu_stall_detector;
+    friend struct signal_handler;
```

Comment:
New friends for reactor class are not welcome. It's large enough already

Reply:
The other ways would be:

1. Add public `reactor::handle_signal_once` and undo deprecation of `reactor::handle_signal`. Users will be able to call them and I think it inverts the purpose of changing the api. Also, it adds more code to the reactor.
2. Refactor signal logic out of the reactor. It seems like the blast radius is big and it's not an easy task.

Thus I would prefer the `friend struct` solution at the moment. Let me know wdyt

Reply:
We already have nested class signal, public nested class signal_handler would be nice replacement for explicit friend. But as I wrote above, installing handler in constructor smells like RAII and implies unsetting it in destructor

Reply:
My goal has been to add a clean seastar namespace level api, not through the reactor.